challenges/it-automation/challenge.md

```diff
@@ -0,0 +1,236 @@
+  sufficient.
+* Avoid sending us draft design documents. It is difficult to evaluate which
+  parts are draft and which parts are complete. Instead we encourage asking
+  questions in Slack and sharing a design document that is ready to be reviewed.
+
+Once the design document has been approved by two reviewers, move on to the
+implementation.
+
+## Implementation
+
+Split your submission into roughly 3-5 Pull Requests to give the team an
+opportunity to review your code and provide feedback. Feel free to merge each
+PR after you have two approvals.
+
+Our team will do their best to provide a high quality review of the submitted
+Pull Requests in a reasonable time frame. You are spending your time on this,
+we are going to contribute our time too.
+
+After the final submission, the interview team will assemble and vote using a
+"+1, -2" anonymous voting system: +1 is submitted whenever a team member
+accepts the submission, -2 otherwise.
+
+In case of a positive result, we will connect you to our HR and recruiting
+teams, who will work out the details and present an offer.
+
+In case of a negative score result, the hiring manager will contact you and
+share a list of the key observations from the team that affected the result.
+
+### Dependencies
+
+Please write as much of your own code as possible. Avoid relying on non-first
+party dependencies. When needed, use reputable sources for key components of the
+challenge.
+
+# Requirements
+
+## Level 3
+
+Use [GitHub Actions](https://github.com/features/actions) to build MVP workflows
+that can do the following.
+
+* Add a workflow that triggers [Terraform](https://www.terraform.io) to
+  configure an Auth0 tenant. Enforce strong authentication for all web
+  applications within tenant.
+* Add a workflow that triggers Terraform to add a web application to the Auth0
+  tenant. Use [OSS Grafana](https://grafana.com/oss) with OIDC authentication
+  as the sample application.
+* Add a workflow that executes a shell script (or Go program) to register
+  users within the Auth0 tenant. Add a sample user to the Auth0 tenant.
+* Securely store and scope all credentials used.
+
+Link to and share successful workflow runs after each PR has been approved and
+merged.
+
+## Level 4
+
+Use [Temporal](https://temporal.io) to build MVP workflows that can do the
+following.
+
+* Create a [Docker Compose](https://docs.docker.com/compose) stack that will
+  be used to run Temporal.
+* Create a CLI tool that can be used to trigger Temporal workflows.
+* Add a workflow that triggers [Terraform](https://www.terraform.io) to
+  configure an Auth0 tenant. Enforce strong authentication for all web
+  applications within tenant.
+* Add a workflow that adds a web application to the Auth0 tenant. Use OSS
+  Grafana with OIDC authentication as the sample application.
+* Add a workflow that registers users within the Auth0 tenant. Add a sample user
+  to the Auth0 tenant.
+* Securely store and scope all credentials used.
```

Comment:
I would like to see tests written too. That's one of the cool things about Temporal that we can test end-to-end workflows.
```suggestion
* Ensure workflows are covered by end-to-end tests.
* Securely store and scope all credentials used.
```